snowpack/src/config.ts

```diff
@@ -539,26 +510,108 @@ export function validatePluginLoadResult(
   }
 }
 
+/**
+ * Get the config base path, that all relative config values should resolve to. In order:
+ *   - The directory of the config file path, if it exists.
+ *   - The config.root value, if given.
+ *   - Otherwise, the current working directory of the process.
+ */
+function getConfigBasePath(configFileLoc: string | undefined, configRoot: string | undefined) {
+  return (
+    (configFileLoc && path.dirname(configFileLoc)) ||
+    (configRoot && path.resolve(process.cwd(), configRoot)) ||
+    process.cwd()
+  );
+}
+
+function resolveRelativeConfigAlias(aliasConfig: Record<string, string>, configBase: string) {
+  for (const [target, replacement] of Object.entries(aliasConfig)) {
+    if (
+      replacement === '.' ||
+      replacement === '..' ||
+      replacement.startsWith('./') ||
+      replacement.startsWith('../') ||
+      replacement.startsWith('/')
+    ) {
+      aliasConfig[target] = target.endsWith('/')
+        ? addTrailingSlash(path.resolve(configBase, replacement))
+        : removeTrailingSlash(path.resolve(configBase, replacement));
+    }
+  }
+}
+
+function resolveRelativeConfigMount(mountConfig: Record<string, any>, configBase: string) {
+  for (const [target, replacement] of Object.entries(mountConfig)) {
+    delete mountConfig[target];
```

Comment:
nit: do we want to mutate the original param? would this cause any problems?

Reply:
yea, good point. The main resolve function is meant to modify the main config object in place (since it's such a potentially complex object, and we're just converting some strings on it to Absolute paths. BUT, since these are just helpers I do like the idea of rewriting them. Will change